tests/Unit/Evolution/Initialization/Test_DgDomain.cpp

```diff
@@ -0,0 +1,500 @@
+struct Component {
+  using metavariables = Metavariables;
+  static constexpr size_t dim = metavariables::dim;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<dim>;
+  using const_global_cache_tags = tmpl::list<domain::Tags::Domain<dim>>;
+
+  using simple_tags =
+      db::AddSimpleTags<domain::Tags::InitialExtents<dim>,
+                        domain::Tags::InitialFunctionsOfTime<dim>, Tags::Time>;
+
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<simple_tags>>>,
+
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<evolution::dg::Actions::InitializeDomain<dim>,
+                     Actions::IncrementTime, Actions::IncrementTime>>>;
+};
+
+template <size_t Dim>
+struct Metavariables {
+  using component_list = tmpl::list<Component<Metavariables>>;
+  static constexpr size_t dim = Dim;
+
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+template <size_t Dim, bool TimeDependent>
+void test() noexcept {
+  using metavars = Metavariables<Dim>;
+  using component = Component<metavars>;
+
+  PUPable_reg(SINGLE_ARG(TimeIndependentMap<Dim, Frame::Inertial>));
+  PUPable_reg(SINGLE_ARG(TimeIndependentMap<Dim, Frame::Grid>));
+  PUPable_reg(SINGLE_ARG(TimeDependentMap<Dim>));
+  PUPable_reg(domain::FunctionsOfTime::PiecewisePolynomial<2>);
+
+  const std::vector<std::array<size_t, Dim>> initial_extents{
+      make_array<Dim>(4_st)};
+  const size_t num_pts = pow<Dim>(4_st);
+  const std::array<double, 3> velocity{{1.2, 0.2, -8.9}};
+  const double initial_time = 0.0;
+  const std::array<std::string, 3> functions_of_time_names{
+      {"TranslationX", "TranslationY", "TranslationZ"}};
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  functions_of_time[functions_of_time_names[0]] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time,
+          std::array<DataVector, 3>{{{0.0}, {velocity[0]}, {0.0}}});
+  functions_of_time[functions_of_time_names[1]] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time,
+          std::array<DataVector, 3>{{{0.0}, {velocity[1]}, {0.0}}});
+  functions_of_time[functions_of_time_names[2]] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time,
+          std::array<DataVector, 3>{{{0.0}, {velocity[2]}, {0.0}}});
+
+  std::vector<Block<Dim>> blocks{1};
+  blocks[0] =
+      Block<Dim>{std::make_unique<TimeIndependentMap<Dim, Frame::Inertial>>(
+                     create_affine_map<Dim, Frame::Inertial>()),
+                 0,
+                 {}};
+  Domain<Dim> domain{std::move(blocks)};
```

Comment:
can the `Block<Dim>` constructor just be put into an initializer list inside this constructor (with some indeterminate number of `{`)?

Reply:
No... C++ is dumb and doesn't do move semantics correctly with initializer lists and so it requires that `blocks` be copyable, which it isn't since it holds `unique_ptr`s

Reply:
Oh, right... Will just told us about that recently. Such a frustrating clash in the standard lib :/